frontend/src/views/SoilTests/SoilTestsModal.tsx

```diff
@@ -90,9 +90,9 @@ export default function SoilTestsModal({
             <div css={{ label: { margin: '0' } }}>
               <StyledDatePicker>
                 <ReactDatePicker
-                  selected={formData.sampleDate}
-                  onChange={(e: any) => {
-                    handleFormFieldChange({ sampleDate: e });
+                  selected={formData.sampleDate ? new Date(formData.sampleDate) : undefined}
```

Comment:
I recommend making a `useMemo` variable for the sampleDate instead of creating a new Date object on every render.

Reply:
Implemented.